### PR TITLE
fix: organize tsconfig

### PR DESCRIPTION
**Pull request by risu729**
### What does this PR do?

This organizes tsconfig for `bun init` and docs.

- Remove `forceConsistentCasingInFileNames` since it is enabled by default. ([docs](https://www.typescriptlang.org/tsconfig#forceConsistentCasingInFileNames))
- Remove `useUnknownInCatchVariables` since it is enabled by default when `strict` is `true`. ([docs](https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables))
- Use consistent tsconfig in `bun init` and docs as follows the [document](https://bun.sh/docs/typescript#:~:text=If%20you%20run%20bun%20init%20in%20a%20new%20directory%2C%20this%20tsconfig.json%20will%20be%20generated%20for%20you.).
  > If you run bun init in a new directory, this tsconfig.json will be generated for you.
- Clarify that the stricter flags are not enabled in `tsconfig.json` generated by `bun init`.

- [x] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### CODE REVIEW

File: docs/guides/runtime/typescript.md [hunk 1]
```diff
@@ -29,15 +29,14 @@ Below is the full set of recommended `compilerOptions` for a Bun project. With t
...
-    // Best practices
+    // Linting
     "strict": true,
     "skipLibCheck": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
     "noFallthroughCasesInSwitch": true,
 
     // Some stricter flags
-    "useUnknownInCatchVariables": true,
```
Comment (necmettindev) [inline docs/guides/runtime/typescript.md:40]: still needs `useUnknownInCatchVariables`

File: docs/typescript.md [hunk 2]
```diff
@@ -31,21 +31,20 @@ Bun supports things like top-level await, JSX, and extensioned `.ts` imports, wh
...
-    // Best practices
+    // Linting
     "strict": true,
     "skipLibCheck": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
     "noFallthroughCasesInSwitch": true,
 
     // Some stricter flags
-    "useUnknownInCatchVariables": true,
```
Comment (necmettindev) [inline docs/typescript.md:42]: and here

File: docs/guides/runtime/typescript.md (hunk 1 above)
Comment (risu729) [reply docs/guides/runtime/typescript.md:40]: It is enabled by default with `strict`, so I think using it is meaningless.

File: src/cli/tsconfig-for-init.json [hunk 3]
```diff
@@ -1,22 +1,22 @@
 {
   "compilerOptions": {
+    // Enable latest features
     "lib": ["ESNext"],
     "target": "ESNext",
     "module": "ESNext",
     "moduleDetection": "force",
-    "jsx": "react-jsx",
-    "allowJs": true,
+    "jsx": "react-jsx", // support JSX
```
Comment (paperclover) [inline src/cli/tsconfig-for-init.json]: is it not obvious setting the tsconfig option `jsx` would enable jsx support.

i dont think this comment adds any value

File: src/cli/tsconfig-for-init.json [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+    "allowJs": true, // allow importing `.js` from `.ts`
```
Comment (paperclover) [inline src/cli/tsconfig-for-init.json]: i dont think this comment is fully accurate or helpful either.

Comment (risu729) [reply src/cli/tsconfig-for-init.json]: I agree with it. I just remained them as is from the previous tsconfig.

File: src/cli/tsconfig-for-init.json [hunk 5]
```diff
@@ -1,22 +1,22 @@
 {
   "compilerOptions": {
+    // Enable latest features
```
Comment (jcarpe) [inline src/cli/tsconfig-for-init.json]: Should we remove comments in this context since it's within JSON?

**Review by jdalton — DISMISSED** [review]
@risu729 Please revert any comment changes and md text changes, keeping the removal of unneeded config options.

**Comment by risu729** [comment]
@jdalton Thank you for your review. I reverted the changes.

> If you run `bun init` in a new directory, this `tsconfig.json` will be generated for you.

My little concern is that `tsconfig.json` in docs and it generated by `bun init` are slightly different.

**Comment by jdalton** [comment]
> My little concern is that `tsconfig.json` in docs and it generated by `bun init` are slightly different.

Ah. We should align them. We can bikeshed that in a separate PR.

**Comment by jdalton** [comment]
Thank you @risu729! 
I'll merge this and if you're up for it a follow-up to align docs with our `bun init` reality would be appreciated.

**Comment by risu729** [comment]
Thank you for merging! I'll open a new PR when this is merged.

**Pull request merged**